src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -943,43 +970,100 @@ protected override void ValidateElement(object element)
                         Metadata.ValidateNotNullFailure);
             }
 
-            var o = element as PSObject;
-            if (o != null)
+            if(_rangeKind.HasValue)
             {
-                element = o.BaseObject;
+                // Using dynamic to handle different type of numbers such as
+                // int, double, decimal and float. 
+                dynamic val = element;
```

Comment:
How do you feel about reusing the ```GetCommonType``` method from the validate range class? 
We could use the incoming object and ```0```. I believe this will also throw the correct message if the user attempts to pass an incompatable object such as string